crates/storage/src/lazy/lazy_array.rs

```diff
@@ -254,30 +216,24 @@ impl<'a, T> DoubleEndedIterator for EntriesIter<'a, T> {
 
 impl<'a, T> ExactSizeIterator for EntriesIter<'a, T> {}
 
-impl<T, N> EntryArray<T, N>
-where
-    N: LazyArrayLength<T>,
+impl<T, const N: usize> EntryArray<T, N>
 {
     /// Creates a new entry array cache.
     pub fn new() -> Self {
         Self {
-            entries: Default::default(),
+            entries:  [(); N].map(|_|Default::default())
```

Comment:
`#![feature(array_map)]` is required to make this work. The issue is that we need to fill the array initially and the array value `CacheCell` does not support `Copy`. So `[Default::default(); N]` does not work.

I tried using our `array-init` dependency, but from my understanding it doesn't support `min_const_generics` yet.